Cesium.CodeGen/Ir/BlockItems/DoWhileStatement.cs

```diff
@@ -0,0 +1,74 @@
+using Cesium.CodeGen.Contexts;
+using Cesium.CodeGen.Extensions;
+using Cesium.CodeGen.Ir.Expressions;
+using Cesium.CodeGen.Ir.Expressions.Constants;
+using Mono.Cecil.Cil;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cesium.CodeGen.Ir.BlockItems;
+
+internal class DoWhileStatement : IBlockItem
+{
+    private readonly IExpression _testExpression;
+    private readonly IBlockItem _body;
+    private readonly string? _breakLabel;
+
+    public DoWhileStatement(Ast.DoWhileStatement statement)
+    {
+        var (testExpression, body) = statement;
+
+        // 6.8.5.3.2 if testExpression is null it should be replaced by nonzero constant
+        _testExpression = testExpression?.ToIntermediate() ?? new ConstantExpression(new IntegerConstant("1"));
```

Comment:
Do we need this conversion in `do … while` loop? It is only specified for the `for` loop. `do … while ();` (without the test expression) doesn't make any sense.